management/src/main/java/io/micronaut/management/health/indicator/threads/OnlyWhenRunningOnTheJvmCondition.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.management.health.indicator.threads;
+
+import io.micronaut.context.condition.Condition;
+import io.micronaut.context.condition.ConditionContext;
+import io.micronaut.core.annotation.Internal;
+
+/**
+ * Condition to hide parts of an application that only work when running on the JVM.
+ * Internal implementation is identical to {@code if (!ImageInfo.inImageCode()).
+ * @author Sergio del Amo
+ * @since 4.8.0
+ */
+@Internal
+class OnlyWhenRunningOnTheJvmCondition implements Condition {
+    private static final String SYS_PROP_ORG_GRAALVM_NATIVEIMAGE_IMAGECODE = "org.graalvm.nativeimage.imagecode";
+    private static final String VALUE_RUNTIME = "runtime";
+    private static final String VALUE_BUILDTIME = "buildtime";
+
+    @Override
+    public boolean matches(ConditionContext context) {
+        return !inImageCode();
+    }
+
+    private static boolean inImageCode() {
+        return inImageBuildtimeCode() || inImageRuntimeCode();
+    }
+
+    private static boolean inImageRuntimeCode() {
+        return VALUE_RUNTIME.equals(System.getProperty(SYS_PROP_ORG_GRAALVM_NATIVEIMAGE_IMAGECODE));
+    }
+
+    private static boolean inImageBuildtimeCode() {
+        return VALUE_BUILDTIME.equals(System.getProperty(SYS_PROP_ORG_GRAALVM_NATIVEIMAGE_IMAGECODE));
+    }
```

Comment:
extract these out to utility methods somewhere so they can be used elsewhere

Reply:
[I extracted a Utils class ](https://github.com/micronaut-projects/micronaut-core/pull/11388/commits/4abec4e94a8a254ab831013a2e7f297956d3d3f5 )